### PR TITLE
*: fix some varcheck lint warnings

### PR DESCRIPTION
**Pull request by Willendless**
### What problem does this PR solve?

Issue Number: #22979 

### What is changed and how it works?

What's Changed:

+ Remove some unused global var
+ Disable linter for some false positive cases

### Check List

Tests

- Unit test

### Release note 

- No release note

### CODE REVIEW

**Review by disksing — APPROVED** [review]

**Comment by disksing** [comment]
Thanks! @Willendless

**Review by nolouch — APPROVED** [review]

File: sessionctx/binloginfo/binloginfo.go [hunk 1]
```diff
@@ -46,6 +46,8 @@ var pumpsClient *pumpcli.PumpsClient
 var pumpsClientLock sync.RWMutex
 var shardPat = regexp.MustCompile(`SHARD_ROW_ID_BITS\s*=\s*\d+\s*`)
 var preSplitPat = regexp.MustCompile(`PRE_SPLIT_REGIONS\s*=\s*\d+\s*`)
+
+// nolint:varcheck
 var autoRandomPat = regexp.MustCompile(`AUTO_RANDOM\s*\(\s*\d+\s*\)\s*`)
```
Comment (tangenta) [inline sessionctx/binloginfo/binloginfo.go:51]: This global variable can be removed.

Comment (Willendless) [reply sessionctx/binloginfo/binloginfo.go]: Got it.

File: executor/hash_table.go [hunk 2]
```diff
@@ -32,18 +32,21 @@ import (
...
 	// estCountMinFactor defines the factor of estCountMin with maxChunkSize.
 	// estCountMin is maxChunkSize * estCountMinFactor, the minimum threshold of estCount.
 	// If estCount is smaller than estCountMin, set estCount to 0.
 	// Set this threshold to prevent buildSideEstCount being too small and causing a performance regression.
 	estCountMinFactor = 8
 
+	// nolint:varcheck
 	// estCountDivisor defines the divisor of buildSideEstCount.
 	// Set this divisor to prevent buildSideEstCount being too large and causing a performance regression.
 	estCountDivisor = 8
```
Comment (zimulala) [inline executor/hash_table.go:52]: Could we remove these constants? Then we don't need `nolint`.

File: domain/infosync/info.go [hunk 3]
```diff
@@ -82,6 +82,7 @@ const (
 // ErrPrometheusAddrIsNotSet is the error that Prometheus address is not set in PD and etcd
 var ErrPrometheusAddrIsNotSet = dbterror.ClassDomain.NewStd(errno.ErrPrometheusAddrIsNotSet)
 
+// nolint:varcheck
 // errPlacementRulesDisabled is exported for internal usage, indicating PD rejected the request due to disabled placement feature.
```
Comment (zimulala) [inline domain/infosync/info.go:86]: Ditto.

File: executor/hash_table.go (hunk 2 above)
Comment (Willendless) [reply executor/hash_table.go]: Sure, I've removed them.

File: ddl/error.go [hunk 4]
```diff
@@ -53,15 +53,16 @@ var (
...
+	// nolint:varcheck
+	errUnknownTypeLength        = dbterror.ClassDDL.NewStd(mysql.ErrUnknownTypeLength)
+	errInvalidDDLJobVersion     = dbterror.ClassDDL.NewStd(mysql.ErrInvalidDDLJobVersion)
+	errInvalidUseOfNull         = dbterror.ClassDDL.NewStd(mysql.ErrInvalidUseOfNull)
+	errTooManyFields            = dbterror.ClassDDL.NewStd(mysql.ErrTooManyFields)
+	errTooManyKeys              = dbterror.ClassDDL.NewStd(mysql.ErrTooManyKeys)
+	errInvalidSplitRegionRanges = dbterror.ClassDDL.NewStd(mysql.ErrInvalidSplitRegionRanges)
+	errReorgPanic               = dbterror.ClassDDL.NewStd(mysql.ErrReorgPanic)
+	errFkColumnCannotDrop       = dbterror.ClassDDL.NewStd(mysql.ErrFkColumnCannotDrop)
+	errFKIncompatibleColumns    = dbterror.ClassDDL.NewStd(mysql.ErrFKIncompatibleColumns)
```
Comment (zimulala) [inline ddl/error.go:65]: Could we remove this code too?

Comment (Willendless) [reply ddl/error.go:65]: Sure, definitely.

File: store/mockstore/unistore/cophandler/mpp.go [hunk 5]
```diff
@@ -44,8 +44,7 @@ const (
 )
 
 const (
-	taskInit int32 = iota
-	taskRunning
+	taskRunning int32 = iota
```
Comment (zimulala) [inline store/mockstore/unistore/cophandler/mpp.go:47]:
```suggestion
	taskRunning int32 = iota +1
```
Maybe using the above code make this logic more secure.

**Pull request merged**